src/arith/iter_affine_map.cc

```diff
@@ -351,8 +374,18 @@ class IterMapRewriter : public ExprMutator {
     std::vector<IterSplitExpr> iters;
     iters.reserve(expr->args.size());
     // canonicalize the expression
+    // find the base scale first
+    PrimExpr base_scale(nullptr);
+    for (const auto& arg : expr->args) {
+      if (is_const_int(arg->scale)) {
+        if (!base_scale.defined() || (CanProveLess(arg->scale, base_scale))) {
```

Comment:
because it is constant, we don't have to do this, just use args->scale.as<IntImmNode> and dig up the values will do